src/analysis/convex/function.lean

```diff
@@ -437,6 +575,67 @@ lemma concave_on.le_right_of_left_le (hf : concave_on 𝕜 s f) {x y z : E} (hx
   f y ≤ f z :=
 @convex_on.le_right_of_left_le 𝕜 E (order_dual β) _ _ _ _ _ _ _ f hf x y z hx hy hz hxz
 
+lemma strict_convex_on.lt_left_of_right_lt' (hf : strict_convex_on 𝕜 s f) {x y : E} (hx : x ∈ s)
+  (hy : y ∈ s) (hxy : x ≠ y) {a b : 𝕜} (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1)
+  (hfy : f y < f (a • x + b • y)) :
+  f (a • x + b • y) < f x :=
+not_le.1 $ λ h, lt_irrefl (f (a • x + b • y)) $
+  calc
+    f (a • x + b • y)
+        < a • f x + b • f y : hf.2 hx hy hxy ha hb hab
+    ... < a • f (a • x + b • y) + b • f (a • x + b • y)
+        : add_lt_add_of_le_of_lt (smul_le_smul_of_nonneg h ha.le) (smul_lt_smul_of_pos hfy hb)
+    ... = f (a • x + b • y) : convex.combo_self hab _
+
+lemma strict_concave_on.left_lt_of_lt_right' (hf : strict_concave_on 𝕜 s f) {x y : E} (hx : x ∈ s)
+  (hy : y ∈ s) (hxy : x ≠ y) {a b : 𝕜} (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1)
+  (hfy : f (a • x + b • y) < f y) :
+  f x < f (a • x + b • y) :=
+@strict_convex_on.lt_left_of_right_lt' 𝕜 E (order_dual β) _ _ _ _ _ _ _ f hf x y hx hy hxy a b ha
+  hb hab hfy
+
+lemma strict_convex_on.lt_right_of_left_lt' (hf : strict_convex_on 𝕜 s f) {x y : E} {a b : 𝕜}
+  (hx : x ∈ s) (hy : y ∈ s) (hxy : x ≠ y) (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1)
+  (hfx : f x < f (a • x + b • y)) :
+  f (a • x + b • y) < f y :=
+begin
+  rw add_comm at ⊢ hab hfx,
+  exact hf.lt_left_of_right_lt' hy hx hxy.symm hb ha hab hfx,
+end
+
+lemma strict_concave_on.lt_right_of_left_lt' (hf : strict_concave_on 𝕜 s f) {x y : E} {a b : 𝕜}
+  (hx : x ∈ s) (hy : y ∈ s) (hxy : x ≠ y) (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1)
+  (hfx : f (a • x + b • y) < f x) :
+  f y < f (a • x + b • y) :=
+@strict_convex_on.lt_right_of_left_lt' 𝕜 E (order_dual β) _ _ _ _ _ _ _ f hf x y a b hx hy hxy ha
+  hb hab hfx
+
+lemma strict_convex_on.lt_left_of_right_lt (hf : strict_convex_on 𝕜 s f) {x y z : E} (hx : x ∈ s)
+  (hy : y ∈ s) (hxy : x ≠ y) (hz : z ∈ open_segment 𝕜 x y) (hyz : f y < f z) :
```

Comment:
Can we drop `(hxy : x ≠ y)` given that we have `(hz : z ∈ open_segment 𝕜 x y)`?

Reply:
Eheh, we actually can because we know that `y ≠ z` because `f y < f z`. Well spotted.

Reply:
Note however that it requires `module 𝕜 E` while those only require `has_scalar 𝕜 E`, so I've duplicated all lemmas.

Reply:
I had not realised `open_segment 𝕜 x x = {x}` when I wrote those words but fortunately you correctly identified the substitute hypothesis `f y < f z`.

Regarding the duplication, personally I'm not yet convinced it's worth it. Couldn't we just have the `module 𝕜 E` versions?

I think it's worth having _definitions_ very general but are people ever going to use these lemmas when they don't have a `module 𝕜 E` structure to hand?

Reply:
As you want. I agree this makes for many lemmas, of which only one or two are currently used (on my branch `sperner-again` I think, actually).